src/convertToObject.js

```diff
@@ -3,16 +3,26 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example in
+ * [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties (see an exampl in
+ * [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .map(value => value.trim())
+    .filter(value => value.length > 0)
+    .map(value => value.split(':'))
+    .reduce((prev, value) => ({
```

Comment:
You made 4 iterations, while it is possible to do in 1. Let's improve algorithm logic.